keps/sig-cluster-lifecycle/kubeadm/20190424-kubeadm-for-windows.md

```diff
@@ -171,58 +183,37 @@ Windows related adjustments to default paths might be required.
+Last, for the paths used, restrictive Access Control Lists (ACL) for Windows should be applied. Golang does not convert Posix permissions to an appropriate Windows ACL, so an additional step is needed. See [mkdirWithACL from moby/moby](https://github.com/moby/moby/blob/e4cc3adf81cc0810a416e2b8ce8eb4971e17a3a3/pkg/system/filesys_windows.go#L103) for an example. This step will be performed by the provisioning script.
 
 #### Kube-proxy deployment
 
-On Linux, kube-proxy is deployed as a DaemonSet in the kubeadm init phase. However, kube-proxy cannot run as a container in Windows since Windows does not support privileged containers. Kube-proxy should therefore be run as a Windows service so that it is restarted by windows control manager automatically and has lifecycle control.
-
-We need to modify the Linux kube-proxy DaemonSet to not deploy on Windows nodes. A PR is already in flight for that [76327](https://github.com/kubernetes/kubernetes/pull/76327). *Merging this PR is mandatory for this proposal*.
-
-Running kube-proxy as a Windows service from kubeadm is out of scope for this proposal. This is due to the fact that we don’t want the changes in kubeadm to be intrusive to the existing method of running kube-proxy as a DaemonSet on Linux. This can end up requiring an abstraction layer that is far from ideal.
-
-The proposed Windows wrapper script that executes kubeadm will also manage the restart of the kube-proxy Windows service.
-
-Long term and ideally, kube-proxy should be run as a DaemonSet on Windows.
+On Linux, kube-proxy is deployed as a DaemonSet in the kubeadm init phase. We will supply a DaemonSet that will run on Windows and use the wins API to launch kube-proxy.
 
 #### CNI plugin deployment
 
-On Linux, CNI plugins are deployed via kubectl and run as a DaemonSet. However, on Windows, CNI plugins need to run on the node, and cannot run in containers (again because Windows does not currently support privileged containers). Azure-CNI, win-bridge (compatible with kubenet), and Flannel all need the binary and config stored on the node.
+On Linux, CNI plugins are deployed via kubectl and run as a DaemonSet. We will supply a DaemonSet that will run Flannel configured in VXLAN/Overlay mode. This ideally can be upstreamed into the Flannel project.
 
-This proposal plans for FlannelD as the default option. Currently, FlannelD has to be started before the kube-proxy Windows service is started. FlannelD creates an HNS network on the Windows host, and kube-proxy will crash if it cannot find the network. This should be fixed in the scope of this project so that kube-proxy will wait until the network comes up. Therefore, kube proxy can be started at any time. 
+If the users wish to use a different plugin they can create a DaemonSet to be applied in place of the Flannel DaemonSet.
```

Comment:
i think for the beta, we might want to document how the users should create a YAML to comply with this setup.
this could be in a `.MD` file in the `sig-windows-tools` repo.

Reply:
Yeah, I am thinking that the Flannel DS will be clear enough for anyone who wants to try to implement a different CNI. Maybe we can add some additional documentation to make clear what's going on in it.